Src/Sarif.PatternMatcher.Cli/AnalyzeDatabaseCommand.cs

```diff
@@ -0,0 +1,302 @@
+using Kusto.Data.Net.Client;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Cli
+{
+    internal class AnalyzeDatabaseCommand : CommandBase
+    {
+        private const int DefaultBatchSize = 10000;
+        private readonly HashSet<string> disabledSkimmers;
+        private readonly ConcurrentDictionary<string, string> content;
+        private bool dynamicValidation;
+        private Channel<string> contentChannel;
+
+        public AnalyzeDatabaseCommand()
+        {
+            disabledSkimmers = new HashSet<string>();
+            content = new ConcurrentDictionary<string, string>();
+        }
+
+        public int Run(AnalyzeDatabaseOptions options)
+        {
+            try
+            {
+                options.Threads = options.Threads > 0 ? options.Threads : Environment.ProcessorCount * 4;
+                dynamicValidation = options.DynamicValidation;
+
+                var channelOptions = new BoundedChannelOptions(int.MaxValue)
+                {
+                    SingleWriter = true,
+                    SingleReader = false,
+                };
+                contentChannel = Channel.CreateBounded<string>(channelOptions);
+
+                AggregatingLogger aggregatingLogger = InitializeLogger(options);
+
+                var sb = new StringBuilder();
+                var writer = new StringWriter(sb);
+                var sarifLogger = new SarifLogger(
+                    writer,
+                    LogFilePersistenceOptions.None,
+                    dataToInsert: options.DataToInsert.ToFlags(),
+                    dataToRemove: options.DataToRemove.ToFlags(),
+                    tool: MakeTool(),
+                    levels: options.Level,
+                    kinds: options.Kind,
+                    closeWriterOnDispose: false);
+                aggregatingLogger.Loggers.Add(sarifLogger);
+
+                aggregatingLogger.AnalysisStarted();
+                ISet<Skimmer<AnalyzeContext>> skimmers = AnalyzeCommand.CreateSkimmersFromDefinitionsFiles(FileSystem, options.SearchDefinitionsPaths);
+
+                var workers = new Task<bool>[options.Threads];
+
+                for (int i = 0; i < options.Threads; i++)
+                {
+                    workers[i] = AnalyzeTargetAsync(aggregatingLogger, skimmers);
+                }
+
+                switch (options.ConnectionType)
+                {
+                    case Enums.ConnectionType.Kusto:
+                    {
+                        QueryKusto(options);
+                        break;
+                    }
+
+                    default:
+                    {
+                        throw new NotImplementedException();
+                    }
+                }
+
+                Task.WhenAll(workers).Wait();
+
+                aggregatingLogger.AnalysisStopped(RuntimeConditions.None);
+                aggregatingLogger?.Dispose();
```

Comment:
>aggregatingLogger?.Dispose(); [](http://example.com/codeflow?start=16&length=29)

we need to dispose manually. Otherwise, .NET won't know that we are still using and will dispose after some time.